app/models/form_responses.rb

```diff
@@ -1,10 +1,30 @@
-class FormResponses < ActiveSupport::HashWithIndifferentAccess
-  def initialize(constructor = nil)
-    self[:id] = SecureRandom.uuid if constructor.nil?
-    super(constructor)
+# Base class for (non-ActiveRecord) models providing serialization and attribute access
+class FormResponses
+  include ActiveModel::Model
+  include ActiveModel::Serialization
+
+  # Derived classes should provide ATTRS (a list of field name symbols), this is
+  # necessary for our serialization process (see #attributes).
+  #
+  # Background https://guides.rubyonrails.org/active_model_basics.html
+  #            https://api.rubyonrails.org/classes/ActiveModel/Model.html
+  #            https://api.rubyonrails.org/classes/ActiveModel/Serialization.html
+
+  ATTRS = [:id].freeze
+  attr_accessor(*ATTRS)
+
+  def initialize(attributes = {})
+    # filter out invalid / no-longer valid fields from data (eg coming from the session)
+    attributes.keep_if { |k, _| self.class::ATTRS.include? k.to_sym }
+    super
+    self.id = SecureRandom.uuid if !self.id
   end
 
-  def id
-    self[:id]
+  def attributes
+    all = Hash.new
+    self.class::ATTRS.each { |attribute| all[attribute] = nil }
```

Comment:
If anyone can tell me how to do this in one line that'd be nice. I spent ages trying to figure out if there was a comprehension-style syntax, and failed :-)

Reply:
`Hash[self.class::ATTRS.map { |attribute| [attribute] }]`

I think that will give you the output:

```
{
  :id  => nil, 
  :date_created => nil,
  :confidence_level_required => nil, 
  :evidence_id_list => nil,
  :identity_exists_over_time => nil,
  :identity_with_organisation_proved => nil,
  :org_check_method => nil,
  :org_activity_found => nil, 
  :activity_time_period => nil,
  :identity_been_stolen_or_used_fraudulently => nil,
  :check_identity_not_stolen_or_used_fraudulently => nil,
  :counter_fraud_data_sources => nil,
  :idependent_counter_fraud_sources => nil
}
```

Reply:
Luuuuverly, thank you